File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-2.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @param {number} k
     * @return {number}
     */
    kthSmallest(root, k) {
        const vals = [];

        this.dfs(root, vals);

        return vals[k - 1]
    }

    dfs(root, arr) {
        if (!root) return;
        this.dfs(root.left, arr);
        arr.push(root.val);
        this.dfs(root.right, arr);

    }

}

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        ans = 0
        def findGoodNodes(root, greatest):
            nonlocal ans
            if not root:
                return
            if root.val >= greatest:
                ans+=1
            new_greatest = max(greatest, root.val)
            findGoodNodes(root.left, new_greatest)
            findGoodNodes(root.right, new_greatest)
        
        findGoodNodes(root, float('-inf'))
        return ans
